packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -170,15 +202,51 @@ export class ConnectionService {
    */
   public async createResponse(
     connectionRecord: ConnectionRecord,
-    outOfBandRecord: OutOfBandRecord
+    outOfBandRecord: OutOfBandRecord,
+    routing?: Routing
   ): Promise<ConnectionProtocolMsgReturnType<ConnectionResponseMessage>> {
     this.logger.debug(`Create message ${ConnectionResponseMessage.type} start`, connectionRecord)
     connectionRecord.assertState(ConnectionState.Requested)
     connectionRecord.assertRole(ConnectionRole.Inviter)
 
+    const { did } = routing ? routing : outOfBandRecord
```

Comment:
Will this work?

```suggestion
    const { did } = routing ?? outOfBandRecord
```

I'm a bit unsure of what this does. In what case is routing undefined, and in what case is the did on the oob record undefined?

Reply:
I somehow ended up creating a new routing when there is a reusable oob invitation, but maybe it's not needed.

Reply:
Do you think we should reuse keys from oob invitation even if it's reusable?